engine/src/cubos/engine/renderer/plugin.cpp

```diff
@@ -101,6 +101,22 @@ static void frameEnvironment(Write<RendererFrame> frame, Read<RendererEnvironmen
     frame->skyGradient(env->skyGradient[0], env->skyGradient[1]);
 }
 
+static void checkPaletteUpdateSystem(Write<Assets> assets, Write<Renderer> renderer,
+                                     Write<ActiveVoxelPalette> activePalette)
+{
+    if (activePalette->asset.isNull())
+    {
+        return;
+    }
+
+    if (assets->update(activePalette->asset) || activePalette->prev != activePalette->asset)
+    {
+        auto palette = assets->read(activePalette->asset).get();
+        (*renderer)->setPalette(palette);
```

Comment:
```suggestion
        auto paletteRead = assets->read(activePalette->asset);
        (*renderer)->setPalette(*paletteRead);
```

One less copy done this way :relieved: